src/kibana/components/index_patterns/_flatten_search_response.js

```diff
@@ -10,7 +10,8 @@ define(function (require) {
         stack.push(key);
         var flattenKey = stack.join('.');
 
-        if ((self.fields.byName[flattenKey] || _.isArray(obj[key]) || !_.isObject(obj[key]))) {
```

Comment:
I'm not sure I understand what is going on here. Can you update this to use [descriptive conditions](https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#use-descriptive-conditions) please?